core/src/main/scala/akka/kafka/scaladsl/KafkaAsyncConsumer.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.kafka.scaladsl
+
+import java.util
+
+import akka.Done
+import akka.actor.{ActorRef, ActorRefFactory}
+import akka.kafka.ConsumerSettings
+import akka.kafka.internal.KafkaConsumerActor
+import akka.util.Timeout
+import org.apache.kafka.clients.consumer.{ConsumerRebalanceListener, ConsumerRecord, OffsetAndMetadata}
+import org.apache.kafka.common.TopicPartition
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.language.postfixOps
+
+object KafkaAsyncConsumer {
+  def apply[K, V](settings: ConsumerSettings[K, V])(implicit as: ActorRefFactory): KafkaAsyncConsumer[K, V] = new KafkaAsyncConsumer[K, V] {
+    import KafkaConsumerActor._
+    import akka.pattern.ask
+    val ref = as.actorOf(KafkaConsumerActor.props(settings))
+    val finished = {
+      implicit val to = Timeout(settings.closeTimeout)
+      Await.result((ref ? Watch).mapTo[Future[Done]], settings.closeTimeout)
+    }
+
+    override def subscribe(topics: Set[String], listener: ConsumerRebalanceListener): Unit = {
+      ref ! Subscribe(topics, listener)
+    }
+
+    override def subscribe(pattern: String, listener: ConsumerRebalanceListener): Unit = {
+      ref ! SubscribePattern(pattern, listener)
+    }
+
+    override def close(): Future[Done] = {
+      as.stop(ref)
+      finished
+    }
+
+    override def request(tps: Set[TopicPartition])(implicit ec: ExecutionContext): Future[Iterator[ConsumerRecord[K, V]]] = {
+      implicit val to = Timeout(1000 hours)
```

Comment:
Help needed here. I have to setup huge timeout here to use ask pattern. Other way is to reply with a future and complete it from actor. What do you think?